File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.go
func topKFrequent(nums []int, k int) []int {
	ans := []int{}
	numMap := map[int]int{}

	for _, num := range nums {
		if _, ok := numMap[num]; ok {
			numMap[num]++
		} else {
			numMap[num] = 1
		}
	}

	for len(ans) < k {
		currentMostFreq := 0
		currentMostFreqAmount := 0
		
		for key, value := range numMap {
			if value > currentMostFreqAmount {
				currentMostFreq = key
				currentMostFreqAmount = value
			}
		}

		ans = append(ans, currentMostFreq)
		delete(numMap, currentMostFreq)
	}

	return ans
}

File: Data Structures & Algorithms/remove-element/submission-1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> ans;
        int result = 0;
        for (const auto& num : nums) {
            if (num != val) {
                ans.push_back(num);
            }
        }

        nums = ans;
        return ans.size();
    }
};